src/composables/accounts.ts

```diff
@@ -222,12 +249,83 @@ export function useAccounts() {
     setActiveAccountByProtocolAndIdx(protocol, account?.idx || 0);
   }
 
-  function setMnemonic(newMnemonic: string) {
+  /**
+   * Setting/Resetting the password key logs the user in/out.
+   */
+  function setPasswordKey(key: IKey | null) {
+    passwordKey.value = key;
+    if (IS_EXTENSION) {
+      if (key) {
+        startSession(key, secureLoginTimeout.value);
+      } else {
+        endSession();
+      }
+    }
+  }
+
+  async function openLoginModal() {
+    setLoaderVisible(true);
+    const sessionKey = await getSessionKey();
+    if (sessionKey) {
+      setPasswordKey(sessionKey);
+      const { getAeSdk } = useAeSdk();
+      await getAeSdk();
+      setLoaderVisible(false);
+      return;
+    }
+    setLoaderVisible(false);
+
+    const { openModal } = useModals();
+
+    await openModal(MODAL_PASSWORD_LOGIN);
+    if (!passwordKey.value) {
+      throw new Error('passwordKey was not set after login.');
+    }
+  }
+
+  async function setPassword(newMnemonic: string, password: string) {
+    const key = await generateKey(password);
+    const encryptedMnemonic = await encrypt(key, newMnemonic);
+    WalletStorage.set(STORAGE_KEYS.mnemonic, encryptedMnemonic);
+
+    // Saved key needs to be generated from the password and the encrypted mnemonic
+    setPasswordKey(await generateKey(password, encryptedMnemonic));
+  }
+
+  async function updatePassword(currentPassword: string, newPassword: string) {
+    const encryptedMnemonic = await WalletStorage.get<string>(STORAGE_KEYS.mnemonic) || undefined;
+    const key = await generateKey(currentPassword, encryptedMnemonic);
+    const decryptedMnemonic = await decrypt(key, encryptedMnemonic!);
+    if (decryptedMnemonic) {
+      await setPassword(decryptedMnemonic, newPassword);
+    } else {
+      throw new Error('Incorrect password.');
+    }
+  }
+
+  async function openSetPasswordModal(newMnemonic: string, isRestored = false) {
+    const { openModal } = useModals();
```

Comment:
I suggest to run the dependant composables at the beginning of the `accounts` composable function.